script.js

```diff
@@ -1,4 +1,259 @@
+    cardTotal += array[i].rank;
+  }
+  return cardTotal;
+}
+
+function newAceTotal(array) {
+  var cardTotal = findCardTotal(array);
+  for (i = 0; i < array.length; i += 1) {
+    if (array[i].rank === 1) {
+      array[i].rank = 11;
+      var index = i;
+      cardTotal = findCardTotal(array);
+      if (cardTotal > 21) {
+        array[index].rank = 1;
+        cardTotal = findCardTotal(array);
+      }
+    }
+  }
+  return cardTotal;
+}
+
+function isMatch(number, cardTotal) {
+  return cardTotal === number;
+}
+
+function isLessThan(number, array) {
+  var cardTotal = newAceTotal(array);
+  return cardTotal < number;
+}
+
+function isBust(number, cardTotal) {
+  //Boolean
+  return cardTotal > number;
+}
+
+function checkWinningMode() {
+  var playerCardTotal = newAceTotal(playerHand);
+  var computerCardTotal = newAceTotal(computerHand);
+  var playerBlackjack = isMatch(21, playerCardTotal);
+  var computerBlackjack = isMatch(21, computerCardTotal);
+  if (isBust(21, playerCardTotal) || isBust(21, computerCardTotal)) {
+    winningMode = "bust";
+  } else if (playerBlackjack || computerBlackjack) {
+    winningMode = "blackjack";
+  } else {
+    if (gameRound === 1) {
+      gameMode = "game continues";
+      gameRound += 1;
+    } else {
+      winningMode = "bigger number";
+    }
+  }
+}
+
+function findWinner() {
+  var playerCardTotal = newAceTotal(playerHand);
+  var computerCardTotal = newAceTotal(computerHand);
+  if (playerCardTotal === computerCardTotal) {
+    var output = "Player and computer";
+  } else if (playerCardTotal > computerCardTotal) {
+    output = "Player";
+  } else {
+    output = "Computer";
+  }
+  return output;
+}
+
+function craftMessage() {
+  var playerCardTotal = newAceTotal(playerHand);
+  var computerCardTotal = newAceTotal(computerHand);
+  if (gameMode === "game continues") {
+    output = `No blackjack, game continues. Player please choose hit or stand! Computer would draw if its hand is below 17<br>`;
+  } else if (gameMode === "find winner") {
+    if (winningMode === "bust") {
+      var busted = findWinner(playerCardTotal, computerCardTotal);
+      output = `${busted} has/have busted. Game restarts<br>`;
+    } else if (winningMode === "blackjack") {
+      var winner = findWinner(playerCardTotal, computerCardTotal);
+      var output = `Blackjack! ${winner} is/are the winner. Game restarts<br>`;
+    } else if (winningMode === "bigger number") {
+      winner = findWinner(playerCardTotal, computerCardTotal);
+      output = `${winner} is/are the winner. Game restarts<br>`;
+    }
+  }
+  return `${output}<br> ${displayAllPlayerCards()}`;
+}
+
+function displayMessage() {
+  checkWinningMode();
+  var playerCardTotal = newAceTotal(playerHand);
+  var computerCardTotal = newAceTotal(computerHand);
+  if (winningMode === "bust") {
+    if (playerCardTotal === computerCardTotal) {
+      output = `Player and computer bust. It's a tie. Game restarts<br>`;
```

Comment:
Similarly, over here, you should declare output above line 155 so that all subsequent lines can use it. It seems that you're doing `output = craftMessage()` many times. Consider writing this line once nearer to the top so that you don't have to keep rewriting it.